v2/audit/conn/internal/writer/writer.go

```diff
@@ -0,0 +1,52 @@
+// Package writer implements a writer to a remote audit server of any type that can be connected to
+// by a net.Conn object.
+package writer
+
+import (
+	"net"
+	"time"
+
+	"github.com/microsoft/go-otel-audit/v2/audit/msgs"
+)
+
+type now func() time.Time
+
+// Conn is a generic writer to a remote audit server. It can use anything that implements the net.Conn interface.
+type Conn struct {
+	conn net.Conn
+	now  now
+}
+
+// Option is an optional argument for the New() function to configure the connection.
+type Option func(c *Conn) error
+
+// New creates a new connection to the remote audit server.
+func New(conn net.Conn, options ...Option) (*Conn, error) {
+	for _, o := range options {
+		if err := o(&Conn{conn: conn, now: time.Now}); err != nil {
```

Comment:
the way you're applying options looks like it might be broken? I think you want


```
func New(conn net.Conn, options ...Option) (*Conn, error) {
        conn := &Conn{conn: conn, now: time.Now}
	for _, o := range options {
		if err := o(conn); err != nil {
			return nil, err
		}
	}
	return conn, nil
}
```